Description improvement and some behavior change for sdr-adjust-gamma

Headcrabed opened the pull request.

Headcrabed commented:
Also I think we could also make `auto` behaves like `yes` on wayland, but I'm not quite sure how to implement this. @Dudemanguy Do you have any idea?

Headcrabed commented on DOCS/man/options.rst:
```diff
@@ -7380,7 +7380,9 @@ them.
-    both source and target metadata is correct.
+    both source and target metadata is correct. If you are using macOS, Windows
+    with ACM enabled, or Wayland that compositor has color management protocol
+    support, metadata should be always correct.
```
Metadata is provided by compositor's color management API, isn't it?

Headcrabed replied:
> This option requires that both source and target metadata is correct. The compositor has nothing to do with the source, it doesn't even know anything about the source file.

Then my opinion is to change this part to “output metadata should be always correct”, is this ok?

llyyr replied:
I prefer saying "On color managed platforms, the output metadata will most likely be correct" instead of the current way of listing gpu-apis and platforms with their settings

Headcrabed commented:
> > > > Also `sdr-adjust-gamma=auto` should be aware of `icc-profile` and `icc-profile-auto`, but I'm not sure how to do this. Any advice?
> > > 
> > > 
> > > What do you mean? icc-profile takes precedence over any other target options.
> > 
> > 
> > So `icc-profile` will be correctly working when `sdr-adjust-gamma=auto`?
> 
> Yes. Is there evidence that would say otherwise?

When I'm testing under windows+ACM off and set `icc-profile`, setting `sdr-adjust-gamma=auto` would result in shift+i shows srgb, while `sdr-adjust-gamma=yes` would result in shift+i shows bt.1886(same as previous version)

llyyr commented on video/out/vo_gpu_next.c:
```diff
@@ -1269,6 +1269,10 @@ static bool draw_frame(struct vo *vo, struct vo_frame *frame)
         case PL_COLOR_TRC_BT_1886:
         case PL_COLOR_TRC_GAMMA22:
         case PL_COLOR_TRC_SRGB:
+#ifdef __APPLE__
```
Color management is also always active on Wayland if the compositor gives us a preferred image description, as well as on Windows with ACM/HDR. This should do the same for all 3 platforms

Headcrabed commented on DOCS/man/options.rst:
```diff
@@ -7380,7 +7380,14 @@ them.
+
+    However, your render API should also support passing output metadata to MPV.
+    Setting ``--gpu-api`` to ``d3d11`` or ``vulkan`` (``opengl`` won't work)
+    would allow this.
```
I changed the description here again, do you think this is ok?

Headcrabed committed:
DOCS/man/options: add some description for sdr-adjust-gamma

Add some description about output metadata.

Signed-off-by: Shengyu Qu <wiagn233@outlook.com>

atrxia1 commented on DOCS/man/options.rst:
```diff
@@ -7380,7 +7380,15 @@ them.
+    management protocol support are all globally color managed.
+
+    However, your render API should also support passing output metadata to MPV,
+    or able to let MPV pick color space description for output surface. Setting
```
It's mpv, not MPV. Please be consistent.

atrxia1 commented on DOCS/man/options.rst:
```diff
@@ -7380,7 +7380,15 @@ them.
+    However, your render API should also support passing output metadata to MPV,
+    or able to let MPV pick color space description for output surface. Setting
+    ``--gpu-api`` to ``d3d11`` or ``vulkan`` (``opengl`` won't work) would allow
+    this.
```
Most of this text seems to be fluff and can be rewritten. Every color management protocol and supported API doesn't need to be listed, these are already mentioned in other places in the manual (e.g. `--target-colorspace-hint` etc.), and don't need to be duplicated. Not mentioning source metadata anywhere in your closing statement also feels weird, because this option was primarily created _because_ source metadata is often more ambiguous than target metadata.

A more accurate and concise summation of this option would be: 

```suggestion
    both source and target metadata is correct. On color-managed platforms,
    the target metadata is likely to be accurate. However, the accuracy of 
    the source metadata cannot be guaranteed.
```

Headcrabed commented:
> I added another commit to partially revert #17008 Since macOS is doing things correctly by default: https://projects.blender.org/blender/blender/issues/145022

Forget about this. My mind wasn't clear. This extra commit is not needed, I've reverted it.

Headcrabed commented:
@kasper93 anything is blocking it from getting merged?

Headcrabed committed:
video/out/vo_gpu_next: don't make sdr-adjust-gamma=auto behave like "no" on macOS

On macOS, global color management is always enabled, so there's no need to
make "auto" behave like "no".

Signed-off-by: Shengyu Qu <wiagn233@outlook.com>

kasper93 commented on video/out/vo_gpu_next.c:
```diff
@@ -1272,6 +1272,10 @@ static bool draw_frame(struct vo *vo, struct vo_frame *frame)
         case PL_COLOR_TRC_GAMMA22:
         case PL_COLOR_TRC_SRGB:
+#ifdef __APPLE__
+            if (opts->sdr_adjust_gamma == 0)
```
> i am also not sure of the necessity of this change

I also don't see why it's needed and how does this differ from other platforms.
